### PR TITLE
[QUIC] Cosmetic changes to Read pipeline

### PR DESCRIPTION
**Pull request by CarnaViire**
Follow-up for NITs and cosmetic changes from #55505

### CODE REVIEW

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs [hunk 1]
```diff
@@ -363,27 +365,38 @@ internal override ValueTask<int> ReadAsync(Memory<byte> destination, Cancellatio
...
+            bool preCanceled = false;
 
             lock (_state)
             {
-                readState = _state.ReadState;
+                initialReadState = _state.ReadState;
                 abortError = _state.ReadErrorCode;
 
-                if (readState != ReadState.PendingRead && cancellationToken.IsCancellationRequested)
+                // Failure scenario: pre-canceled token. Transition: any -> Aborted
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs:377]: Love the comments :heart:

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs [hunk 2]
```diff
@@ -363,27 +365,38 @@ internal override ValueTask<int> ReadAsync(Memory<byte> destination, Cancellatio
...
+                // PendingRead state indicates there is another concurrent read operation in flight
+                // which is forbidden, so it is handled separately
+                if (initialReadState != ReadState.PendingRead && cancellationToken.IsCancellationRequested)
                 {
-                    readState = ReadState.Aborted;
+                    initialReadState = ReadState.Aborted;
                     _state.ReadState = ReadState.Aborted;
-                    canceledSynchronously = true;
+                    preCanceled = true;
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs:384]: Why don't directly return `new OperationCanceledException(cancellationToken)` here and get rid of `preCanceled` local var. We wouldn't even need to change the `initialReadState` if I'm reading the code correctly.

**Review by ManickaP — APPROVED** [review]
LGTM, thanks!

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs (hunk 2 above)
Comment (CarnaViire) [reply src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs:384]: To do all the work on exceptions in one place and outside of the lock

**Pull request merged**